### ISSUE TITLE
Invalid pattern parsing in nested "or" pattern

### ISSUE THREAD

**Issue by CohenArthur** [Rust-GCC/gccrs#1808]
I tried this code:

```rust
struct C(char);

fn main() {
    let c = C('a');
    match c {
        C('a' | 'b') => {}
        _ => {}
    }
}

```

I expected to see this happen: no error

Instead, this happened: 
```rust
test.rs:6:15: error: unexpected token ‘|’ in tuple struct items
    6 |         C('a' | 'b') => {}
      |        
```
  
### Meta

- What version of Rust GCC were you using, 7f222689dcb

**Comment by ArberSephirotheca** [comment]
Hi, I am interested in this issue and currently looking at the related code, could you please assign me if possible?

**Comment by P-E-P** [comment]
Fixed by #1878
